x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorInvalidStatusTransitionException.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+public class ConnectorInvalidStatusTransitionException extends Exception {
+
+    /**
+     * Constructs an ConnectorInvalidStatusTransitionException exception with a detailed message.
+     *
+     * @param current The current state of the connector.
+     * @param next The attempted next state of the connector.
+     */
+    public ConnectorInvalidStatusTransitionException(ConnectorStatus current, ConnectorStatus next) {
+        super(
+            "Invalid transition attempt from ["
+                + current
+                + "] to ["
+                + next
+                + "]. Such a status transition is not supported by the Connectors Protocol."
```

Comment:
Nit: unnecessary plural (we just call it Connector Protocol)

```suggestion
                + "]. Such a status transition is not supported by the Connector Protocol."
```